OpenSim/Actuators/Millard2012AccelerationMuscle.cpp

```diff
@@ -1536,93 +1464,83 @@ SimTK::Vector Millard2012AccelerationMuscle::
         iter++;
     }
 
-    //*******************************    
-    //Populate the output vector
     //*******************************
-    //If the solution converged
-    if(abs(ddlce_dtt) < aSolTolerance){    
-        //1: flag (0 = converged
-        //         1 = diverged (not enough iterations) 
-        //         2 = no solution due to singularity:length 0, 
-        //         3 = no solution due to pennation angle singularity
-        //2: solution Error (N)
-        //3: iterations
-        //4: fiber length (m)
-        //5: fiber velocity (m/s)
-        //6: passive force (N)
-        //7: tendon force (N)
-        
-        results[0] = 0;
-        results[1] = ddlce_dtt;
-        results[2] = (double)iter;
-        results[3] = lce;
-        results[4] = dlce_dt;
-        results[5] = ami.fpeVEM * fiso;
-        results[6] = ami.fseVEM * fiso;
-    }else{ //If the solution diverged
-        
-        //Fiber length is at its lower bound
-        if(iter < aMaxIterations){
+    // Populate the result vector
+    //
+    //   [0] solution error (N)
+    //   [1] iterations
+    //   [2] fiber length (m)
+    //   [3] fiber velocity (m/s)
+    //   [4] passive force (N)
+    //   [5] tendon force (N)
+    //*******************************
+    if (abs(ddlce_dtt) < aSolTolerance) {  // The solution converged.
+
+        result[0] = ddlce_dtt;
+        result[1] = (double)iter;
+        result[2] = lce;
+        result[3] = dlce_dt;
+        result[4] = ami.fpeVEM * fiso;
+        result[5] = ami.fseVEM * fiso;
+        return ResultOfInitMuscleState::Success_Converged;
+    }
```

Comment:
Maybe consider returning an associative array instead of a sequence if values represent different quantities. The associative array will improve readability.
```
results["solution_error"]
results["fiber_velocity"]
.....so on
```
If you have another reason to return a sequence, consider `std::vector` instead of `SimTK::Vector`.